### PR TITLE
Refactor SQLA models `UUID` type hints and add type annotations to account_limit

### PR DESCRIPTION
**Pull request by rdimaio**
Part of #6588. The `UUID` type hints are changed to `str` per discussion with Dimitrios

### CODE REVIEW

File: lib/rucio/gateway/account_limit.py [hunk 1]
```diff
@@ -56,16 +66,22 @@ def get_local_account_limits(account, vo='def', *, session: "Session"):
     :returns: The account limits.
     """
 
-    account = InternalAccount(account, vo=vo)
+    internal_account = InternalAccount(account, vo=vo)
```
Comment (voetberg) [inline lib/rucio/gateway/account_limit.py:69]: This rename should probably be a different commit (same goes for lines 98, 123, 152 etc)

Comment (rdimaio) [reply lib/rucio/gateway/account_limit.py:69]: It's directly related to the type annotation change (because by specifying `account: str`, the linter complains about this line as we're now passing an `InternalAccount` to an object that should be `str`), but I'll break these out in a separate commit for clarity

**Comment by rdimaio** [comment]
Rebased in latest forcepush

File: lib/rucio/gateway/account_limit.py (hunk 1 above)
Comment (rdimaio) [reply lib/rucio/gateway/account_limit.py:69]: Addressed in latest forcepush

**Review by bari12 — APPROVED** [review]
